.changeset/beige-ants-pay.md

```diff
@@ -0,0 +1,5 @@
+---
+"@remix-run/router": patch
+---
+
+Fix issue with fetcher data cleanup in the data layer on fetcher unmount
```

Comment:
Your PR is targeting `main` instead of `dev`, so the merge has added a bunch of extra changes like this. Could you update the base branch for this PR?

Reply:
Oops, initially I thought I was supposed to base my PRs off of `main`, but now I know. Changed base.